### PR TITLE
DBMS-assignment-3 SQL Constraints, Functions & Views

### PR DESCRIPTION
**Pull request by meta-hitesh-pareek**
Assignment 1:
With respect to StoreFront application identify, apply and list the constraints you would apply on the columns for the tables created.

Assignment 2:
Write SQL scripts for the following:
Display the list of products (Id, Title, Count of Categories) which fall in more than one Categories.
Display Count of products as per below price range:

Range in Rs. | Count
-- | --
0 - 100 |  
101 - 500 |  
Above 500 |  
Display the Categories along with number of products under each category.

Assignment 3:
Write SQL scripts for the following:
Display Shopper’s information along with number of orders he/she placed during last 30 days.
Display the top 10 Shoppers who generated maximum number of revenue in last 30 days.
Display top 20 Products which are ordered most in last 60 days along with numbers.
Display Monthly sales revenue of the StoreFront for last 6 months. It should display each month’s sale.
Mark the products as Inactive which are not ordered in last 90 days.
Given a category search keyword, display all the Products present in this category/categories. 
Display top 10 Items which were cancelled most.

Assignment 4:
Consider a form where providing a Zip Code populates associated City and
State. 
Create appropriate tables and relationships for the same and write a SQL
         query for that returns a Resultset containing Zip Code, City Names and
         States ordered by State Name and City Name.

Assignment 5:
Create a view displaying the order information (Id, Title, Price, Shopper’s name, Email, Orderdate, Status) with latest ordered items should be displayed first for last 60 days.
Use the above view to display the Products(Items) which are in ‘shipped’ state.
Use the above view to display the top 5 most selling products.

### CODE REVIEW

File: DBMS_Assignment3_Question2.sql [hunk 1]
```diff
@@ -0,0 +1,49 @@
+
+/* QuestiON 1*/
```
Comment (sharma-mohit-eca) [inline DBMS_Assignment3_Question2.sql:2]: Before submitting please review by your own.

File: DBMS_Assignment3_Question2.sql [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+GROUP BY product_category.product_id;
+
+/*QuestiON 2*/
+SELECT 
+    '0-100' AS 'Range_in_rs.', count(*) AS count
+FROM
+    product
+WHERE
+    product_price > 0 AND product_price <= 100
+GROUP BY 'Range-in-rs.'UNION SELECT
```
Comment (sharma-mohit-eca) [inline DBMS_Assignment3_Question2.sql:20]: Spacing issues

File: DBMS_Assignment3_Question2.sql (hunk 1 above)
Comment (sharma-mohit-eca) [reply DBMS_Assignment3_Question2.sql:2]: Paste the brief of the sql query as comment

File: DBMS_Assignment3_Question4.sql [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+);
+
+CREATE TABLE zipcodes(
+    zipcode int primary key not null,
+    city_no int not null,
+    foreign key (city_no)
+        references cities (city_no)
+);
+
+INSERT INTOstates(state_name) VALUES ("Rajasthan");
```
Comment (sharma-mohit-eca) [inline DBMS_Assignment3_Question4.sql:27]: no spacing

File: DBMS_Assignment3_Question2.sql (hunk 2 above)
Comment (meta-hitesh-pareek) [reply DBMS_Assignment3_Question2.sql:20]: Sir Mysql Workbench is giving auto spacing when I use the shortcut ctrl+b;

**Pull request merged**